### PR TITLE
TableExport IR

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

**Comment by jigold** [comment]
I wasn't 100% sure what the output for Pretty should look like. I tried to follow what was done for other IRs.

File: src/main/scala/is/hail/expr/ir/Pretty.scala [hunk 1]
```diff
@@ -89,6 +89,9 @@ object Pretty {
                 s"$path overwrite"
               else
                 path
+            case TableExport(_, path, typesFile, header, exportType) =>
+              val args = Array(Some(path), Option(typesFile), if (header) Some("header") else None, Some(exportType)).flatten
```
[the remark this answers is not in the document]
Comment (tpoterba) [reply src/main/scala/is/hail/expr/ir/Pretty.scala]: we should write our own scala "repr"

[the remark this answers is not in the document]
Comment (tpoterba) [reply src/main/scala/is/hail/expr/ir/Pretty.scala]: we basically have this for `prettyIdentifier`

[the remark this answers is not in the document]
Comment (danking) [reply src/main/scala/is/hail/expr/ir/Pretty.scala]: Ah, I didn't know about that. Great! Let's use `StringEscapeUtils.escapeString(path, sb, backticked=false`

[the remark this answers is not in the document]
Comment (danking) [reply src/main/scala/is/hail/expr/ir/Pretty.scala]: What do you mean by our own scala "repr"? Do you mean a string representation of the IR that we can import and export?

**Comment by jigold** [comment]
This is what is printed right now. I think if we want to do something drastically different, it should be in a different PR and we should decide what we want the standard to be.

```
(TableExport
  file:/tmp/hail.Suywz2yoz6a3/out.00000..tsv
  header
  0
  (TableLiteral))
```

**Pull request merged**